src/Markdown.ts

```diff
@@ -185,7 +205,7 @@ export default class Markdown {
                                  * but this solution seems to work well and is hopefully slightly easier to understand too
                                  */
                                 const format = formattingChangesByNodeType[node.type];
-                                const nonEmphasizedText = `${format}${node.firstChild.literal}${format}`;
+                                const nonEmphasizedText = `${format}${innerNodeLiteral(node)}${format}`;
```

Comment:
The node is unlinked here https://github.com/matrix-org/matrix-react-sdk/pull/9570/files#diff-3080504149f1c5e77feab2433cd58ec989ca431665891e39534fa4969b43eb01R220

Current implementation assumes the text being in a single node inside the emphasis. If it has more levels it will drop everything except the contents of the first child node.

To fix this we need to get the entire „inner“ text of the node being dropped.

I hope my thoughts here are correct. The markdown processing is kind of complex here and hard to understand.